scaleway/helpers.go

```diff
@@ -275,15 +274,28 @@ func regionSchema() *schema.Schema {
 		allRegions = append(allRegions, z.String())
 	}
 	return &schema.Schema{
-		Type:        schema.TypeString,
-		Description: "The region you want to attach the resource to",
-		Optional:    true,
-		ForceNew:    true,
-		Computed:    true,
-		ValidateFunc: validation.StringInSlice(
-			allRegions,
-			true,
-		),
+		Type:             schema.TypeString,
+		Description:      "The region you want to attach the resource to",
+		Optional:         true,
+		ForceNew:         true,
+		Computed:         true,
+		ValidateDiagFunc: validateWithWarnings(allRegions, "region"),
+	}
+}
+
+// validateWithWarnings helps to only returns warnings in case we got a non public locality passed
+func validateWithWarnings(correctValues []string, field string) func(i interface{}, path cty.Path) diag.Diagnostics {
```

Comment:
`validateLocality` ?

Reply:
The code is not specific to Locality so it was unclear whether we might need it for something else.

Reply:
`validateStringInSliceWithWarning` ?